src/OctopusPuppet.Tests/DeploymentPlanActionTests.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using OctopusPuppet.DeploymentPlanner;
+using OctopusPuppet.OctopusProvider;
+using OctopusPuppet.Tests.TestHelpers;
+using Xunit;
+
+namespace OctopusPuppet.Tests
+{
+    [Collection("NonParallelCollection")]
+    public class DeploymentPlanActionTests
+    {
+        private const string EnvironmentId = "D1";
+
+        private static TestComponent[] GetComponents() => new[]
+        {
+            new TestComponent { ProjectName = "ArmSharedInfrastructure", Version = "1.2.3456" },
+            new TestComponent { ProjectName = "Filebeat", Version = "2.3.456" },
+        };
+         
+
+        [Theory]
+        [InlineData("1.2.3456")]
+        [InlineData("2.3.456")]
+        public void GetBranchDeploymentPlan_Should_Skip_MasterBuilds_When_SkipNoBranchSuffix_IsTrue(string branch)
```

Comment:
These two tests verify what happens to non-suffixed releases when `SkipNoBranch` is set. However, we have no test to verify that `SkipNoBranch` has no impact on branches with a suffix.